### ISSUE TITLE
Visible marker to distinguish required/optional tool params

### ISSUE THREAD

**Issue by elichad** [galaxyproject/galaxy#14627]
**Describe the bug**
Tool params that are marked up with `optional=true` aren't distinguished from non-optional params in the UI. 

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.01 (muongalaxy.stfc.ac.uk) and 22.05 (usegalaxy.eu)

**Expected behavior**
There should be a clear distinction between required and optional fields in the tool form UI (both visible and screen reader accessible).

There are a couple of common ways this is done elsewhere on the web:
- asterisk on required fields (usually red) 
- text '(optional)' after optional fields (often smaller and/or greyed-out)

The best option for Galaxy probably depends on whether the majority of fields across all tools are optional or required.

--

**Screenshots**
The second and fourth fields in this screenshot are optional, but there's no way to tell.
![Screenshot of tool form containing four fields with identical formatting](https://user-images.githubusercontent.com/20683271/190468292-32ea283b-47d2-498f-922d-9367c3480a07.png)

**Comment by nsoranzo** [comment]
I think the red asterisk is the universally known indicator for required fields, so I'd vote for that independently of the proportion of optional/required fields, which varies among tools (and tool developers' "style").

An additional note: this may not be needed for certain parameter types, e.g. `boolean`, which are never optional, but maybe that would be confusing?

**Comment by patrick-austin** [comment]
Related to this, when using an `optional` `select` parameter, it would be useful to be able to define the text that appears on the default option (that corresponds to a value of `None`). Currently this is "Nothing selected". In our case we want to have an optional selector for file extensions, and if none is selected no output file is generated. But in context "Nothing selected" doesn't accurately convey this, being able to have something like the second option in the list replace "Nothing selected" would be clearer for a user.

![image](https://user-images.githubusercontent.com/61705287/190600884-112233c8-d5af-4e04-b48d-4e24551ce43f.png)

**Comment by elichad** [comment]
@patrick-austin that might be better as a separate issue as I think it would also involve changes to the tool XML schema, not just the web UI

**Comment by nsoranzo** [comment]
> @patrick-austin that might be better as a separate issue as I think it would also involve changes to the tool XML schema, not just the web UI (also to avoid having two separate conversations about design in the same issue)

@elichad Agreed, I've created a separate issue referencing @patrick-austin's comment.
